src/brigadier_components/string_reader.ts

```diff
@@ -89,6 +94,34 @@ export class StringReader {
         this.cursor += str.length;
         return helper.succeed();
     }
+    /**
+     * Require that a specific string of a set of strings follows
+     * @param opts The options for the expectation
+     */
+    public expectOption(opts: string[]): ReturnedInfo<string> {
```

Comment:
This can be generic in the style of `readOption`. That allows more strongly typed uses of the result.